src/components/application_manager/test/request_tracker_test.cc

```diff
@@ -0,0 +1,250 @@
+
+    EXPECT_CALL(mock_request_controller_settings_,
+                app_hmi_level_none_time_scale_max_requests())
+        .WillRepeatedly(ReturnRef(kDefaultAppHmiLevelNoneTimeScaleMaxRequests));
+
+    EXPECT_CALL(mock_request_controller_settings_, app_time_scale())
+        .WillRepeatedly(ReturnRef(kDefaultAppRequestsTimeScale));
+
+    EXPECT_CALL(mock_request_controller_settings_,
+                app_time_scale_max_requests())
+        .WillRepeatedly(ReturnRef(kDefaultAppTimeScaleMaxRequests));
+  }
+
+ protected:
+  application_manager_test::MockRequestControlerSettings
+      mock_request_controller_settings_;
+
+  application_manager::request_controller::RequestTracker tracker_;
+
+  const uint32_t kDefaultAppHmiLevelNoneRequestsTimeScale = 10;
+  const uint32_t kDefaultAppHmiLevelNoneTimeScaleMaxRequests = 100u;
+  const uint32_t kDefaultAppTimeScaleMaxRequests = 5;
+  const uint32_t kDefaultAppRequestsTimeScale = 200u;
+};
+
+TEST_F(RequestTrackerTestClass, TrackAppRequestInNone_ExpectSuccessTillLimit) {
+  const uint32_t app_id = 1;
+  const mobile_apis::HMILevel::eType none_level =
+      mobile_apis::HMILevel::HMI_NONE;
+
+  SetDefaultConstraints();
+
+  for (size_t i = 0; i < kDefaultAppHmiLevelNoneTimeScaleMaxRequests; ++i) {
+    EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+              tracker_.Track(app_id, none_level));
+  }
+
+  EXPECT_EQ(application_manager::request_controller::TrackResult::
+                kNoneLevelMaxRequestsExceeded,
+            tracker_.Track(app_id, none_level));
+}
+
+TEST_F(RequestTrackerTestClass,
+       TrackAppRequestInNone_NoLimits_ExpectAlwaysSuccess) {
+  const uint32_t app_id = 1;
+  const mobile_apis::HMILevel::eType none_level =
+      mobile_apis::HMILevel::HMI_NONE;
+
+  const uint32_t no_limit = 0;
+
+  EXPECT_CALL(mock_request_controller_settings_,
+              app_hmi_level_none_time_scale())
+      .WillRepeatedly(ReturnRef(no_limit));
+
+  EXPECT_CALL(mock_request_controller_settings_,
+              app_hmi_level_none_time_scale_max_requests())
+      .WillRepeatedly(ReturnRef(no_limit));
+
+  for (size_t i = 0; i < kDefaultAppHmiLevelNoneTimeScaleMaxRequests; ++i) {
+    EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+              tracker_.Track(app_id, none_level));
+  }
+
+  EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+            tracker_.Track(app_id, none_level));
+}
+
+TEST_F(RequestTrackerTestClass,
+       TrackAppRequestInOtherThanNone_ExpectSuccessTillLimit) {
+  const uint32_t app_id = 1;
+  mobile_apis::HMILevel::eType hmi_level =
+      mobile_apis::HMILevel::HMI_BACKGROUND;
+
+  SetDefaultConstraints();
+
+  for (size_t i = 0; i < kDefaultAppTimeScaleMaxRequests; ++i) {
+    EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+              tracker_.Track(app_id, hmi_level));
+    if (i % 2) {
```

Comment:
@AByzhynar 
This is easy way to use different values of HMI level in test without complicated logic.